django/test/client.py

```diff
@@ -833,6 +833,8 @@ def _handle_redirects(self, response, data='', content_type='', **extra):
                 data = QueryDict(url.query)
                 content_type = None
 
+            if response.request['REQUEST_METHOD'] != 'GET':
+                path = urljoin(path, '?' + url.query)
```

Comment:
As far as I'm concerned this should impact only `307` and `308` redirects, so maybe:
```diff
diff --git a/django/test/client.py b/django/test/client.py
index b26504f762..e4201bead4 100644
--- a/django/test/client.py
+++ b/django/test/client.py
@@ -827,7 +827,10 @@ class Client(ClientMixin, RequestFactory):
 
             if response.status_code in (HTTPStatus.TEMPORARY_REDIRECT, HTTPStatus.PERMANENT_REDIRECT):
                 # Preserve request method post-redirect for 307/308 responses.
-                request_method = getattr(self, response.request['REQUEST_METHOD'].lower())
+                request_method = response.request['REQUEST_METHOD'].lower()
+                if request_method != 'get':
+                    extra['QUERY_STRING'] = url.query
+                request_method = getattr(self, request_method)
             else:
                 request_method = self.get
                 data = QueryDict(url.query)
```

Reply:
Agreed. I originally had it nested within the 307, 308 if block but moved it out